### PR TITLE
feat(FR-2300): add CSP frame-src diagnostic check

### PR DESCRIPTION
**Pull request by agatha197**
Resolves #5966 (FR-2300)

## Summary
- Add `checkCspFrameSrc` diagnostic rule that checks whether the API endpoint origin is allowed by CSP `frame-src` (falls back to `default-src` per CSP spec)
- Integrate the new frame-src check into `useCspDiagnostics` hook, emitting a warning when blocked or a passed result when allowed
- Add comprehensive unit tests for `checkCspFrameSrc` covering empty inputs, wildcard, fallback to default-src, and blocking scenarios
- Add i18n keys for frame-src diagnostic messages (`CspFrameSrcBlocked`, `CspFrameSrcPassed`, and related description/fix keys)

## Changed Files
- `react/src/diagnostics/rules/cspRules.ts` — new `checkCspFrameSrc` function
- `react/src/diagnostics/rules/__tests__/cspRules.test.ts` — tests for `checkCspFrameSrc`
- `react/src/hooks/useCspDiagnostics.ts` — integrate frame-src check
- `resources/i18n/en.json` — add frame-src diagnostic i18n keys

### CODE REVIEW

File: resources/i18n/en.json [hunk 1]
```diff
@@ -646,6 +646,11 @@
     "CspApiEndpointBlocked": "API Endpoint Blocked by CSP",
     "CspApiEndpointBlockedDesc": "The API endpoint ({{endpoint}}) is not included in the Content-Security-Policy connect-src directive.",
     "CspApiEndpointBlockedFix": "Add {{endpoint}} to the connect-src directive in your CSP meta tag or server headers.",
+    "CspFrameSrcBlocked": "API Endpoint Blocked from iframe Embedding by CSP",
+    "CspFrameSrcBlockedDesc": "The API endpoint ({{endpoint}}) is not included in the Content-Security-Policy frame-src directive. Iframe-based features may not work correctly.",
+    "CspFrameSrcBlockedFix": "Add {{endpoint}} to the frame-src directive in your CSP meta tag or server headers.",
+    "CspFrameSrcPassed": "CSP frame-src OK",
+    "CspFrameSrcPassedDesc": "CSP frame-src allows embedding {{endpoint}} in iframes.",
```
[the remark this answers is not in the document]
Comment (agatha197) [reply resources/i18n/en.json]: Translations for non-English locales are handled in a separate i18n batch workflow. These keys will be translated when the feature set stabilizes.

File: resources/i18n/en.json [hunk 2]
```diff
@@ -646,6 +646,11 @@
     "CspApiEndpointBlocked": "API Endpoint Blocked by CSP",
     "CspApiEndpointBlockedDesc": "The API endpoint ({{endpoint}}) is not included in the Content-Security-Policy connect-src directive.",
     "CspApiEndpointBlockedFix": "Add {{endpoint}} to the connect-src directive in your CSP meta tag or server headers.",
+    "CspFrameSrcBlocked": "API Endpoint Blocked from iframe Embedding by CSP",
```
[the remark this answers is not in the document]
Comment (agatha197) [reply resources/i18n/en.json]: Good point. Fixed the capitalization to "Iframe Embedding" in the latest commit.

File: resources/i18n/en.json [hunk 3]
```diff
@@ -646,6 +646,11 @@
     "CspApiEndpointBlocked": "API Endpoint Blocked by CSP",
     "CspApiEndpointBlockedDesc": "The API endpoint ({{endpoint}}) is not included in the Content-Security-Policy connect-src directive.",
     "CspApiEndpointBlockedFix": "Add {{endpoint}} to the connect-src directive in your CSP meta tag or server headers.",
+    "CspFrameSrcBlocked": "API Endpoint Blocked from Iframe Embedding by CSP",
```
Comment (ironAiken2) [inline resources/i18n/en.json:649]: It seems that the translations for other languages are missing.

Comment (agatha197) [reply resources/i18n/en.json:649]: Translations for all supported languages are added in the stacked PR #5980.

File: react/src/diagnostics/rules/cspRules.ts [hunk 4]
```diff
@@ -264,3 +264,44 @@ export function checkCspStyleSrc(
...
+export function checkCspFrameSrc(
+  cspContent: string | null | undefined,
+  apiEndpoint: string,
+  pageOrigin?: string,
+): DiagnosticResult | null {
+  if (!cspContent || !apiEndpoint) return null;
+
+  const sources = getEffectiveSources(cspContent, 'frame-src');
+  if (!sources) return null;
+
```
[the remark this answers is not in the document]
Comment (agatha197) [reply react/src/diagnostics/rules/cspRules.ts:282]: You're correct that per the CSP spec, the full fallback chain is `frame-src` → `child-src` → `default-src`. However, `child-src` for frames is deprecated in CSP Level 3, and modern browsers use `frame-src` directly. For diagnostics purposes, the `frame-src` → `default-src` fallback covers the practical cases. Adding `child-src` support could be a future enhancement.

File: react/src/diagnostics/rules/__tests__/cspRules.test.ts [hunk 5]
```diff
@@ -320,3 +321,79 @@ describe('checkCspStyleSrc', () => {
...
+
+    // default-src blocks the endpoint — should warn
+    const result = checkCspFrameSrc(
+      'default-src https://other.example.com',
+      'https://api.example.com',
+    );
+    expect(result).not.toBeNull();
+    expect(result?.severity).toBe('warning');
+  });
+
```
[the remark this answers is not in the document]
Comment (agatha197) [reply react/src/diagnostics/rules/__tests__/cspRules.test.ts:383]: As noted in the reply above, `child-src` fallback for frames is deprecated in CSP Level 3. The current test coverage for `frame-src` and `default-src` fallback is sufficient for the practical use cases.
